docs/DeveloperGuide.md

```diff
@@ -256,47 +261,269 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
+
+**MSS:**
+
+1. User will enter the keyword that he want to search for
+2. The system will display a list of tasks that contain the keyword.
+
+Use case ends
+
+**Extension:**
+
+1a. User enters no keyword.
+> The system will display "Invalid command format" along with the format of findCommand.
+
+Use case ends
+
+### Use case: Find tasks - filter by day/week
+
+**MSS:**
+
+1. User will enter the day/week he want to look at using CLI.
+2. The System will display a list of tasks that are related(start/due on) to the day/week.
+
+Use case ends
+
+**Extension:**
+
+1a. User enter a input that is not a number
+> The System will display "please enter a valid number when search by day/week".
+1b. User will enter the wrong syntax for find command
+> The system will display "Invalid command format" along with the format of findCommand.
+
+Use case ends
+
+### Use case : Delete a task (one entry)
+
+**MSS:**
+
+1. User will enter the index of the task to be deleted using CLI.
+2. System will display "Deleted Task(s) : TASKNAME".
+
+Use case ends
+
+**Extensions:**
+
+1a. User enter a non-integer value or invalid number.
+> The System will display "Invalid command format!" along with the format of the delete command
+
+1b. User enter an invalid index.
+> System will display 'the following indexes are invalid and ignore: INDEXES' and proceed to delete the valid index found.
+
+Use case ends
+
+### Use case : Delete tasks (multiple entries)
+
+**MSS:**
+
+1. User will enter mutiple indexs of tasks to be deleted using CLI
+2. System will display "Deleted Task(s): INDEXES"
+
+Use case ends
+
+**Extensions:**
+
+1a. User enter a non-integer value.
+> The System will display "Invalid command format!" along with the format of the delete command.
+
+1b. User enter an invalid index.
+> System will display 'the following indexes are invalid and ignore: INDEXES' and proceed to delete the valid index found.
+
+Use case ends
+
+
+### Use case : Editing a task (edit)
+
+**MSS:**
+
+1. User will enter the index of the tasks to be edited along with new data.
+2. System will display "Edit saved!".
 
-#### Use case: Delete person
+Use case ends
 
-**MSS**
+**Extensions:**
+
+1a. Index provided by user is not valid
+>The system will display an error message along with the suggested format for the edit command
 
-1. User requests to list persons
-2. AddressBook shows a list of persons
-3. User requests to delete a specific person in the list
-4. AddressBook deletes the person <br>
 Use case ends.
 
-**Extensions**
+1b. Not all details are provided by the user.
+>The system will only edit the details of the task the user provides and leave the rest unmodified.
 
-2a. The list is empty
+Use case ends.
 
-> Use case ends
+1c. The user did not enter any new details
+>The system will leave the task unmodified
 
-3a. The given index is invalid
+Use case ends.
+
+
+### Use case : Clear task list (clear)
+
+**MSS:**
+
+1. User will enter clear to clear the save data.
```

Comment:
Here what is the difference between delete and clear, Is it the "save data" here mention a temporary stored data?
Because it seems this simple command executing by mistake can result in loosing all data. Um just curious. If it does not temporary data I suggest you to have some pre-steps to ensure user really want to clear saved data